docs/tutorials/nextjs.mdx

```diff
@@ -0,0 +1,186 @@
+---
+id: nextjs
+title: NextJS
+sidebar_label: NextJS
+hide_title: true
+---
+
+&nbsp;
+
+# NextJS Application Tutorial
+
+:::tip What You'll Learn
+
+- How to set up and use Redux Toolkit with the NextJS framework
+
+:::
+
+:::info Prerequisites
+
+- Familiarity with [ES6 syntax and features](https://www.taniarascia.com/es6-syntax-and-feature-overview/)
+- Knowledge of React terminology: [JSX](https://reactjs.org/docs/introducing-jsx.html), [State](https://reactjs.org/docs/state-and-lifecycle.html), [Function Components, Props](https://reactjs.org/docs/components-and-props.html), and [Hooks](https://reactjs.org/docs/hooks-intro.html)
+- Understanding of [Redux terms and concepts](https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow)
+- Working through the [Quick strt guide](./quick-start.mdx) is recommended
+
+:::
+
+## Introduction
+
+NextJS is a server side rendering framework for React that presents some unique challenges for using Redux properly. There are two architectures for a NextJS application; the Pages Router and the App Router. The Pages Router is the original architecture for NextJS. Using Redux with the Pages Router is well understood and handled primarily by the (next-redux-wrapper)[https://github.com/kirill-konshin/next-redux-wrapper]. This tutorial will focus on the App Router architecture as it is the new default architecture option for NextJS and it presents some unique challenges for using Redux properly.
+
+### How to Read This Tutorial
+
+This page assumes that you already have an exisiting NextJS application based on the App Router architecture.
+
+## The App Router Architecture and Redux
+
+The primary new feature of the NextJS App Router is the addition of support for React Server Components (RSCs). RSCs are a special type of React component that only renders on the server, as opposed to "client" components that render on **both** the client and the server. RSCs can be defined as `async` functions and return promises during rendering as they make async requests for data to render.
+
+RSCs abilitiy to block for data means that with the App Router you no longer have `getServerSideProps` to fetch data for rendering. Any component in the tree can make asychronous requests for data. While this is very convenient it also means thats if you define global variables (like the Redux store) they will be shared across requests. This is a problem because the Redux store could be contaminated with data from other requests.
```

Comment:
💬 yep, the "shared across requests" bit is huge.

Reply:
I'll just point out that that is also true for the `pages` directory - there it's just not as much of a common problem, because `getServerSideProps` and `next-redux-wrapper` give you a pattern that doesn't involve globals. But if you use them, you're equally screwed.